### DISCUSSION TITLE
HIGHLIGHTS: Summarizing Replays

### DISCUSSION THREAD

**Discussion by nirlipo** [pacman-ai-projects/pacman-contest-cluster#102] [Ideas]
At the XAI workshop of Explainable AI, I saw an invited talk by Ofra Amir (Harvard) about automated summarization of MDP agents playing Pacman. 

It's a cool elegant idea that can be used to extract the `highlights` of each game! 

I guess we have a new feature request for the contest scripts & Dashboard!

[HIGHLIGHTS: Summarizing Agent Behavior to People](https://scholar.harvard.edu/files/oamir/files/highlightsmain.pdf)

[Link to the code](https://github.com/HuTobias/HIGHLIGHTS-LRP)

> ABSTRACT
People increasingly interact with autonomous agents. This paper
introduces and formalizes the problem of automatically generating a summary of an agent’s behavior with the goal of increasing
people’s familiarity with the agent’s capabilities and limitations.
In contrast with prior approaches which developed methods for
explaining a single decision made by an agent, our approach aims to
provide users with a summary that describes the agent’s behavior
in different situations. We hypothesize that reviewing such summaries could help people in tasks such as choosing between agents
or determining the level of autonomy to grant to an agent. We
develop “HIGHLIGHTS”, an algorithm that produces a summary of
an agent’s behavior by extracting important trajectories from simulations of the agent. We conducted a human-subject experiment to
evaluate whether HIGHLIGHTS summaries help people assess the
capabilities of agents. Our results show that participants were more
successful at evaluating the capabilities of agents when presented
with HIGHLIGHTS summaries compared to baseline summaries,
and rated them as more helpful. We also explore a variant of the
HIGHLIGHTS algorithm which aims to increase the diversity of
states included in the summary, and show that this modification
further improves people’s ability to assess agents’ capabilities.

**Comment by AndrewPaulChester** [comment]
This seems really interesting, unfortunately as it stands it requires a reasonable q-function for states. It doesn't seem to matter if the q-function is the agent's own, so in theory if we had a good q-function then we could run it (not sure if there is an existing example of a good RL agent that goes directly from observations to actions). 

The fact that we would need to use an off-policy q-function also suggests a potential modification to the algorithm. Instead of defining importance as maxQ - minQ for a particular state, we could define it as maxQ - chosenQ, which would highlight the states where the agent made mistakes (according to our q-function). Otherwise my intuition is that the important states will be crowded by ones where the agent is running away from a ghost, which may not convey much useful information if the agent is moderately competent.

**Comment by nirlipo** [comment]
Yes, you are right. Most RL agents implemented by students used hand-crafted features. We could reuse a high performing agent q-function from previous competitions.
